### PR TITLE
ENH: Update noxfile and env

### PR DESCRIPTION
**Pull request by trallard**
This PR adds the following

- Enhancements to noxfile to use conda and streamline installs
- Uses environment.yaml instead of requirements.txt
- Small docs

### CODE REVIEW

File: tests/retrolab/environment.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  - nodejs=16.*
+  - yarn
+  - matplotlib
+  - ipywidgets
+  - ipython
+  - numpy
+  - scipy 
+  - pip
+  - pip:
+    - git+https://github.com/jupyterlab/retrolab.git@v0.3.19#egg=retrolab
```
Comment (gabalafou) [inline tests/retrolab/environment.yml]: this is what I was looking for! :)

File: tests/retrolab/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+## :zap: Running the tests
+
+1. Make sure you are in the correct directory:
+
+    ```bash
+    cd tests/retrolab
+    ```
+
+2. Run the test with snox from the command line:
```
Comment (gabalafou) [inline tests/retrolab/README.md]:
```suggestion
2. Run the test with nox from the command line:
```

now you've got me wondering what a snox would be if a snox were a thing... sounds like something from Dr. Seuss or Roald Dahl might come up with...

File: tests/retrolab/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Retrolab accessibility tests
+
+## :package: Requirements
+
+To run the tests in this directory you need the following pre-requisites:
+
+- python > 3.7
```
Comment (gabalafou) [inline tests/retrolab/README.md:8]: mamba?

```suggestion
- mamba
- python > 3.7
```

File: tests/retrolab/noxfile.py [hunk 4]
```diff
@@ -1,13 +1,39 @@
...
+
+# global variables
+FILE = Path(__file__)
+TEST_DIR = FILE.parent
+ENV_FILE = TEST_DIR / "environment.yml"
+
+# parse the env file
+environment = safe_load(ENV_FILE.read_text())
+dependencies = environment.get("dependencies")
+requirements = dependencies.pop(-1).get("pip")
```
Comment (gabalafou) [inline tests/retrolab/noxfile.py:15]: oh... you cheated :-P

File: tests/retrolab/noxfile.py [hunk 5]
```diff
@@ -1,13 +1,39 @@
...
+
+
+def install_environment(session):
+    """Install conda dependencies - instead of creating a conda env through
+    conda env create -f we install depency into the session venv.
+    This is faster than recreating, updating or pruning the environment using
```
Comment (gabalafou) [inline tests/retrolab/noxfile.py:21]: oh why is that? (faster)

File: tests/retrolab/noxfile.py [hunk 6]
```diff
@@ -1,13 +1,39 @@
...
+
+# parse the env file
+environment = safe_load(ENV_FILE.read_text())
+dependencies = environment.get("dependencies")
+requirements = dependencies.pop(-1).get("pip")
+
+
+def install_environment(session):
+    """Install conda dependencies - instead of creating a conda env through
+    conda env create -f we install depency into the session venv.
```
Comment (gabalafou) [inline tests/retrolab/noxfile.py]:
```suggestion
    conda env create -f we install dependency into the session venv.
```

File: tests/retrolab/README.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+1. Make sure you are in the correct directory:
+
+    ```bash
+    cd tests/retrolab
+    ```
+
+2. Run the test with snox from the command line:
+
+    ```bash
+    nox -s a11y_tests
```
Comment (gabalafou) [inline tests/retrolab/README.md:29]: but if they don't have mamba, this will fail won't it?

**Review by gabalafou — CHANGES_REQUESTED** [review]

File: tests/retrolab/README.md (hunk 2 above)
Comment (trallard) [reply tests/retrolab/README.md]: 😆  just a typo but would be a nice name for another tool to add to the tox/nox confusion

File: tests/retrolab/README.md (hunk 3 above)
Comment (trallard) [reply tests/retrolab/README.md:8]: of course 😮‍💨

File: tests/retrolab/noxfile.py (hunk 4 above)
Comment (trallard) [reply tests/retrolab/noxfile.py:15]: yes - can talk about why - but long story short nox does not play as nicely with conda as it does with venv

File: tests/retrolab/README.md (hunk 7 above)
Comment (trallard) [reply tests/retrolab/README.md:29]: yup - maybe can add a check for mamba or conda and allow for the system to set the default 🤔  actually will do that

File: tests/retrolab/noxfile.py (hunk 5 above)
Comment (trallard) [reply tests/retrolab/noxfile.py:21]: part of the same story as above

File: tests/retrolab/README.md (hunk 7 above)
Comment (gabalafou) [reply tests/retrolab/README.md:29]: great! will be on the lookout for your update

**Comment by gabalafou** [comment]
Thanks @trallard! That Dockerfile looks like it was a lot of work. I'm going to merge this in because I want to build on top of it 🚀

**Pull request merged**